src/compas/utilities/itertools.py

```diff
@@ -185,10 +189,79 @@ def linspace(start, stop, num=50):
 
 
 def flatten(list_of_lists):
-    """Flatten one level of nesting"""
+    """Flatten one level of nesting.
+
+    Examples
+    --------
+    >>> a = [[1, 2, 3], [4, 5, 6]]
+    >>> list(flatten(a))
+    [1, 2, 3, 4, 5, 6]
+    """
     return chain.from_iterable(list_of_lists)
 
 
+def unflatten(lst, num):
+    """Returns a nested list.
+
+    Parameters
+    ----------
+    lst : list
+        A list of items.
+    num : int
+        The length of the sub-list.
+
+    Raises
+    ------
+    ValueError
+        If the length of the list is not a factor of num.
+
+
+    Examples
+    --------
+    >>> a = [1, 2, 3, 4, 5, 6]
+    >>> unflatten(a, 3)
+    [[1, 2, 3], [4, 5, 6]]
+    """
+    if len(lst) % num:
+        raise ValueError("The length of the array must be a factor of n: %d %% %d == 0" % (len(lst), num))
+    return [lst[i:i + num] for i in range(0, len(lst), num)]
+
+
+def reshape(lst, shape):
+    """Gives a new shape to an array without changing its data.
+
+    This function mimicks the functionality of ``numpy.reshape`` [1]_, but in a simpler form.
+
+    Parameters
+    ----------
+    lst : list
+        A list of items.
+    shape : int or tuple of ints
```

Comment:
I don't think it makes sense for `shape` to be an `int`.  As it is now, that would result in a `TypeError`, but I don't think it really makes sense to call `reshape([1,2,3], 3)`.  It would make sense if the input were something more complicated than a flat list.  You could address this by using the version of `flatten` suggested above and then recursively calling a helper function which does what `reshape` currently does, but maybe that's more than you need:
```
def reshape(l, shape):
    def helper(l, shape):
        if len(shape) == 1:
            if len(l) % shape[0] != 0:
                raise ValueError()
            return [l[i:i + shape[0]] for i in range(0, len(l), shape[0])]
        n = reduce(mul, shape[1:])
        return [helper(lst[i * n:(i + 1) * n], shape[1:]) for i in range(len(lst) // n)]

    shape = (shape,) if instance(shape, int) else shape
    l = flatten(l)
    if len(l) != reduce(lambda x, y: x * y, shape):
        raise ValueError("ValueError: cannot reshape array of size %d into shape %s" % (len(lst), shape))
    return helper(l, shape)

Reply:
thank you for suggesting even better utility functions ;)